### PR TITLE
Add note of arguments evaluation timing

### PR DESCRIPTION
**Pull request by MakeNowJust**
I think this is confuse!

### CODE REVIEW

File: _gitbook/syntax_and_semantics/macros/macro_methods.md [hunk 1]
```diff
@@ -20,3 +20,19 @@ person.instance_vars_names #=> ["name", "age"]
 
 Note that in the case of macro defs you need to specify the return type.
 
+Arguments in macro defs are evaluated in running time, not a compile time:
```
Comment (jhass) [inline _gitbook/syntax_and_semantics/macros/macro_methods.md]: It's true for all macro's though, mentioning it only for macro defs could only replace it with another misconception :)

Comment (MakeNowJust) [inline _gitbook/syntax_and_semantics/macros/macro_methods.md]: That's right.  My intended is  "Arguments passed to macro defs method call are not passing macro expansion, but passing to expanded method".

Comment (jhass) [inline _gitbook/syntax_and_semantics/macros/macro_methods.md]: Maybe we should just include distinction a bit more explicit here by repeating the behaviour for macros, how about:

> In macro definitions arguments are passed as their AST nodes, giving you access to them in macro expansions (`{{some_macro_argument}}`). However that is not true for macro defs, here the argument list is that of the method generated by the macro def, you cannot not access their compile time value.

>  ```cr
class Object
  macro def has_instance_var?(name) : Bool
    # We cannot access name inside the macro expansion here, instead we need to use the
    # macro language to construct an array and do the inclusion check at runtime.
    return {{ @type.instance_vars.map &.name.stringify }}.includes? name
  end
end

Comment (jhass) [inline _gitbook/syntax_and_semantics/macros/macro_methods.md]: Which reminds me that I actually still would love something like a macro def that generates a new method for each distinct set of arguments.

Comment (asterite) [inline _gitbook/syntax_and_semantics/macros/macro_methods.md]: I agree that docs should mention this distinction. The way @jhass said it is good.

Comment (MakeNowJust) [inline _gitbook/syntax_and_semantics/macros/macro_methods.md]: I agree too.  Could @jhass commit docs added your said? Then, I will close this pull request ;)

Comment (jhass) [inline _gitbook/syntax_and_semantics/macros/macro_methods.md]: I don't mind either way, will open a PR.

**Pull request closed without merging**